libs/providers/in-memory/src/lib/in-memory-provider.ts

```diff
@@ -32,51 +34,70 @@ export class InMemoryProvider implements Provider {
   }
 
-  async resolveStringEvaluation(flagKey: string): Promise<ResolutionDetails<string>> {
-    const flagValue = this.lookupFlagValueOrThrow(flagKey);
+  async resolveObjectEvaluation(flagKey: string, defaultValue: JsonValue, context?: EvaluationContext, logger?: Logger): Promise<ResolutionDetails<any>> {
+    return Promise.resolve(this.resolveFlagWithReason<JsonValue>(flagKey, defaultValue, context, logger));
+  }
 
-    if (typeof flagValue !== 'string') {
-      throw new TypeMismatchError();
+  private resolveFlagWithReason<T extends JsonValue | FlagValueType>(flagKey: string, defaultValue: T, ctx?: EvaluationContext, logger?: unknown): ResolutionDetails<any>{
+    try {
+      const resolutionResult = this.lookupFlagValueOrThrow(flagKey, ctx, logger)
+
+      if (typeof resolutionResult?.value != typeof defaultValue) {
+        throw new TypeMismatchError();
+      }
+
+      return resolutionResult;
+    } catch (error: unknown) {
+      return {
+        value: defaultValue, reason: error instanceof DisabledError ?
+          StandardResolutionReasons.DISABLED
+          : StandardResolutionReasons.DEFAULT
+      }
     }
-
-    return staticResolution(flagValue);
   }
 
-  async resolveNumberEvaluation(flagKey: string): Promise<ResolutionDetails<number>> {
-    const flagValue = this.lookupFlagValueOrThrow(flagKey);
-
-    if (typeof flagValue !== 'number') {
-      throw new TypeMismatchError();
+  private lookupFlagValueOrThrow(flagKey: string, ctx?: EvaluationContext, logger?: unknown): ResolutionDetails<any>{
+    if (!(flagKey in this._flagConfiguration)) {
+      throw new FlagNotFoundError();
     }
+    const flagSpec: Flag = this._flagConfiguration[flagKey];
 
-    return staticResolution(flagValue);
-  }
+    this.throwIfFlagDisabled(flagSpec);
+
+    const isContextEval = ctx && flagSpec?.contextEvaluator;
+    const variant = isContextEval ? flagSpec.contextEvaluator?.(ctx) : flagSpec.defaultVariant;
+
+    const value = variant && flagSpec?.variants[variant];
+
+    if (value === undefined) {
+      throw new TargetingKeyMissingError();
+    }
 
-  async resolveObjectEvaluation<U extends JsonValue>(flagKey: string): Promise<ResolutionDetails<U>> {
-    throw new GeneralError('support for object flags has not been implemented');
+    return { value, ...(variant && { variant }), reason: isContextEval ? StandardResolutionReasons.TARGETING_MATCH : StandardResolutionReasons.STATIC, };
   }
 
-  private lookupFlagValueOrThrow(flagKey: string): FlagValue {
-    if (!(flagKey in this._flagConfiguration)) {
-      throw new FlagNotFoundError();
+  private throwIfFlagDisabled(flagSpec: Flag) {
+    if (flagSpec.disabled) {
+      throw new DisabledError();
     }
-    return this._flagConfiguration[flagKey];
   }
 }
 
-function staticResolution<U>(value: U): ResolutionDetails<U> {
-  return {
-    value,
-    reason: StandardResolutionReasons.STATIC,
-  };
+class DisabledError extends Error {
```

Comment:
Perhaps we should add this as a standard error type. @toddbaert What do you think?